### PR TITLE
Change feedback api to trpc 

### PR DESCRIPTION
**Pull request by Troff8**
## PR includes

- [x] Bug Fix

## Related issues

Resolve [#1323 ](https://github.com/taskany-inc/issues/issues/1323)

### CODE REVIEW

File: trpc/context.ts [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
 import { getServerSession } from 'next-auth/next';
+import { IncomingHttpHeaders } from 'http';
 
 import { authOptions } from '../src/utils/auth';
 
 export const createContext = async (opts: trpcNext.CreateNextContextOptions) => {
     const session = await getServerSession(opts.req, opts.res, authOptions);
 
-    return { session };
+    return { session, headers: opts.req.headers as IncomingHttpHeaders };
```
Comment (9teen90nine) [inline trpc/context.ts:11]: Can't `opts.req` be undefined?

File: trpc/router/feedback.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import { protectedProcedure, router } from '../trpcBackend';
+import { createFeedbackSchema } from '../../src/schema/feedback';
+
+export const feedback = router({
+    create: protectedProcedure.input(createFeedbackSchema).mutation(async ({ ctx, input }) => {
+        if (!process.env.FEEDBACK_URL) {
+            return;
+        }
+        const { title, description, href } = input;
```
Comment (awinogradov) [inline trpc/router/feedback.ts:9]: You can do inline on 5 line, like `({ ctx, input: { title, description, href }})`

File: src/utils/declareSsrProps.ts [hunk 3]
```diff
@@ -39,7 +39,14 @@ export function declareSsrProps<T = ExternalPageProps>(
             };
         }
 
-        const ssrHelpers = createServerSideHelpers({ router: trpcRouter, ctx: { session }, transformer: superjson });
+        const ssrHelpers = createServerSideHelpers({
+            router: trpcRouter,
+            ctx: {
+                session,
+                headers: {},
```
Comment (9teen90nine) [inline src/utils/declareSsrProps.ts:46]: why empty? req is also available here

File: trpc/context.ts (hunk 1 above)
Comment (Troff8) [reply trpc/context.ts]: i removed as IncomingHttpHeaders

File: src/schema/schema.i18n/ru.json [hunk 4]
```diff
@@ -1,20 +1,19 @@
...
-    "Goal's title is required": "",
-    "Goal's title must be a string": "",
-    "Goal's description must be longer than 10 symbols": "",
-    "Goal's description is required": "",
-    "Goal's description must be a string": "",
-    "Goal's project or team are required": "",
-    "Criteria weight is required": "",
-    "Criteria weight must be longer than 1 symbol": "",
-    "Choose a dependency": ""
+    "Title is required": "Требуется заголовок",
```
Comment (awinogradov) [inline src/schema/schema.i18n/ru.json:2]: Заголовок обязателен

File: src/schema/schema.i18n/ru.json [hunk 5]
```diff
@@ -1,20 +1,19 @@
...
+    "Title must be a string": "Заголовок должен быть строкой",
+    "Title must be longer than {upTo} symbol": "Заголовок должен быть длиннее {upTo} символов",
+    "Comments's description is required": "Требуется описание комментариев",
```
Comment (awinogradov) [inline src/schema/schema.i18n/ru.json:5]: Текст комментария обязателен

File: src/schema/schema.i18n/ru.json [hunk 6]
```diff
@@ -1,20 +1,19 @@
...
+    "Comments's description must be a string": "Описание комментариев должно быть строкой",
```
Comment (awinogradov) [inline src/schema/schema.i18n/ru.json:6]: Текст комментария может быть только строкой

File: src/schema/schema.i18n/ru.json [hunk 7]
```diff
@@ -1,20 +1,19 @@
...
+    "Comments's description must be longer than 1 symbol": "Описание комментариев должно быть длиннее 1 символа",
```
Comment (awinogradov) [inline src/schema/schema.i18n/ru.json:7]: Текст комментария должен быть длиннее 1 символа

File: src/schema/schema.i18n/ru.json [hunk 8]
```diff
@@ -1,20 +1,19 @@
...
+    "Title must be longer than 2 symbols": "Заголовок должен быть длиннее 2 символов",
+    "Title can be 50 symbols maximum": "Заголовок может содержать не более 50 символов",
+    "Goal's title is required": "Требуется название цели",
```
Comment (awinogradov) [inline src/schema/schema.i18n/ru.json:10]: Заголовок обязателен — also, I think you can reuse some duplicates. Like this one same with key on 2 line.

File: src/schema/schema.i18n/ru.json [hunk 9]
```diff
@@ -1,20 +1,19 @@
...
+    "Goal's title must be a string": "Название цели должно быть строкой",
```
Comment (awinogradov) [inline src/schema/schema.i18n/ru.json:11]: Заголовок может быть только строкой

File: src/schema/schema.i18n/ru.json [hunk 10]
```diff
@@ -1,20 +1,19 @@
...
+    "Goal's description must be longer than 10 symbols": "Описание цели должно быть длиннее 10 символов",
+    "Goal's description is required": "Требуется описание цели",
```
Comment (awinogradov) [inline src/schema/schema.i18n/ru.json:13]: Описание цели обязательно

File: src/schema/schema.i18n/ru.json [hunk 11]
```diff
@@ -1,20 +1,19 @@
...
+    "Goal's description must be a string": "Описание цели должно быть строкой",
```
Comment (awinogradov) [inline src/schema/schema.i18n/ru.json:14]: Описание может быть только строкой

File: src/schema/schema.i18n/ru.json [hunk 12]
```diff
@@ -1,20 +1,19 @@
...
+    "Goal's project or team are required": "Требуется проект или команда цели",
```
Comment (awinogradov) [inline src/schema/schema.i18n/ru.json:15]: Укажите проект

Also look at #1338

File: src/schema/schema.i18n/ru.json [hunk 13]
```diff
@@ -1,20 +1,19 @@
...
+    "Criteria weight is required": "Требуется вес критериев",
```
Comment (awinogradov) [inline src/schema/schema.i18n/ru.json:16]: Укажите вес критерия

File: src/schema/schema.i18n/ru.json [hunk 14]
```diff
@@ -1,20 +1,19 @@
...
+    "Criteria weight must be longer than 1 symbol": "Вес критерия должен быть длиннее 1 символа",
+    "Choose a dependency": "Выберите зависимость"
```
Comment (awinogradov) [inline src/schema/schema.i18n/ru.json:18]: Укажите тип зависимости

File: src/schema/schema.i18n/ru.json [hunk 15]
```diff
@@ -1,20 +1,19 @@
...
+    "Title must be longer than 2 symbols": "Заголовок должен быть длиннее 2 символов",
+    "Title can be 50 symbols maximum": "Заголовок может содержать не более 50 символов",
+    "Goal's title is required": "Требуется название цели",
+    "Goal's title must be a string": "Название цели должно быть строкой",
+    "Goal's description must be longer than 10 symbols": "Описание цели должно быть длиннее 10 символов",
+    "Goal's description is required": "Требуется описание цели",
+    "Goal's description must be a string": "Описание цели должно быть строкой",
+    "Goal's project or team are required": "Требуется проект или команда цели",
+    "Criteria weight is required": "Требуется вес критериев",
+    "Criteria weight must be longer than 1 symbol": "Вес критерия должен быть длиннее 1 символа",
```
Comment (awinogradov) [inline src/schema/schema.i18n/ru.json:17]: I think this key is incorrect. It's possible to have first criteria with 99 weight and second with 1 weight

cc @LamaEats

**Review by awinogradov — CHANGES_REQUESTED** [review]

File: src/schema/schema.i18n/en.json [hunk 16]
```diff
@@ -1,20 +1,18 @@
...
+    "Comments's description is required": "Comments's description is required",
+    "Comments's description must be a string": "Comments's description must be a string",
+    "Comments's description must be longer than 1 symbol": "Comments's description must be longer than 1 symbol",
+    "Title must be longer than 2 symbols": "Title must be longer than 2 symbols",
+    "Title can be 50 symbols maximum": "Title can be 50 symbols maximum",
+    "Goal's title must be a string": "Goal's title must be a string",
+    "Goal's description must be longer than 10 symbols": "Goal's description must be longer than 10 symbols",
+    "Goal's description is required": "Goal's description is required",
+    "Goal's description must be a string": "Goal's description must be a string",
+    "Goal's project are required": "Goal's project are required",
```
Comment (awinogradov) [inline src/schema/schema.i18n/en.json:14]: is required

File: src/schema/schema.i18n/ru.json (hunk 9 above)
Comment (awinogradov) [reply src/schema/schema.i18n/ru.json:10]: ?

Comment (Troff8) [reply src/schema/schema.i18n/ru.json:10]: this key was written like that, I just translated it, I don’t know that the title can only be a string

File: src/schema/schema.i18n/ru.json (hunk 15 above)
Comment (Troff8) [reply src/schema/schema.i18n/ru.json:16]: I didn't understand🙃 
what should be written instead of this key?

**Review by awinogradov — APPROVED** [review]

**Review by 9teen90nine — APPROVED** [review]

**Pull request merged**